### PR TITLE
more closely model GPX file format, add waypoints

### PR DESCRIPTION
**Pull request by chetbox**

### CODE REVIEW

File: GpxKit/Gpx+Parser.swift [hunk 1]
```diff
@@ -24,12 +32,22 @@ public extension Gpx {
...
+        return Metadata(name: name, description: description, author: author)
+    }
+
     func point() throws -> Point {
         guard let element = element else { throw IndexingError.error }
         let latitude = try element.attributeValue(by: "lat")
         let longitude = try element.attributeValue(by: "lon")
-        return Point(coordinate: CLLocationCoordinate2D(latitude: latitude, longitude: longitude))
+        let time = self["time"].element?.text
+        return Point(coordinate: CLLocationCoordinate2D(latitude: latitude, longitude: longitude), time: time != nil ? TimeInterval(time!) : nil)
```
Comment (marcbaldwin) [inline GpxKit/Gpx+Parser.swift:48]: You might be able to do `time: time as? TimeInterval)`

File: GpxKit/Gpx+Parser.swift [hunk 2]
```diff
@@ -24,12 +32,22 @@ public extension Gpx {
 
 fileprivate extension XMLIndexer {
 
+    func metadata() -> Metadata? {
+        guard element != nil else { return nil }
```
Comment (marcbaldwin) [inline GpxKit/Gpx+Parser.swift:36]: Does Xcode give a warning here that element is unused? Why do you need the check?

Comment (chetbox) [reply GpxKit/Gpx+Parser.swift:36]: There may not be a `<metadata>` element in the XML in which case `element` here is `nil`. Indexing into a document with SWXMLHash gives you a reference to an element that may or may not exist. This simply returns early. You're right, we don't need it. Shall I remove it?

File: GpxKit/Gpx+Parser.swift (hunk 1 above)
Comment (chetbox) [reply GpxKit/Gpx+Parser.swift:48]: This does't work. :( `TimeInterval` is a number and `time` is a `String`. We have to call the constructor to parse the string, I think.

File: GpxKit/Gpx+Parser.swift (hunk 2 above)
Comment (marcbaldwin) [reply GpxKit/Gpx+Parser.swift:36]: Um nah that seems fine 👍

**Pull request merged**